### PR TITLE
Mining with memory bank

### PR DESCRIPTION
**Pull request by AlekseySh**

### CODE REVIEW

File: oml/configs/miner/miner_with_bank.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+name: miner_with_bank
+args:
+  bank_size_in_batches: 3
+  need_logs: True
+  miner:
+    name: nhard_triplets
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/configs/miner/miner_with_bank.yaml:6]: okay, renamed

File: oml/miners/miner_with_bank.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        assert isinstance(bank_size_in_batches, int)
+        assert bank_size_in_batches >= 1
+        assert isinstance(miner, NHardTripletsMiner)
+        self.miner = miner
+
+        self.bank_size_in_batches = bank_size_in_batches
+        self.bank_features: Optional[Tensor] = None
+        self.bank_labels: Optional[Tensor] = None
+        self.bank_size = -1
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/miners/miner_with_bank.py:42]: added

File: oml/miners/miner_with_bank.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+            bs = features.shape[0]
+            self.feat_dim = features.shape[-1]
+            self.bank_size = self.bank_size_in_batches * bs
+            self.bank_labels = torch.empty(self.bank_size, dtype=torch.long, device=features.device)
+            self.bank_features = torch.empty(
+                (self.bank_size, self.feat_dim), dtype=features.dtype, device=features.device
+            )
+
+    @no_grad()
+    def update_bank(self, features: Tensor, labels: Tensor) -> None:
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/miners/miner_with_bank.py:63]: agree, I'll make it private

File: oml/miners/miner_with_bank.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                ids_a=ids_a, ids_p=ids_p, ids_n=ids_n, ignore_anchor_mask=ignore_anchor_mask
+            )
+
+        self.update_bank(features=features, labels=labels)
+
+        return features_miner[ids_a], features_miner[ids_p], features_miner[ids_n]
+
+    @staticmethod
+    def _prepare_logs(
+        ids_a: List[int], ids_p: List[int], ids_n: List[int], ignore_anchor_mask: Tensor
```
[the remark this answers is not in the document]
Comment (AlekseySh) [reply oml/miners/miner_with_bank.py:114]: added

**Review by b0nce — APPROVED** [review]

**Pull request merged**